File: src/main/java/com/duxtinto/intellij/plugin/github/codereviews/ui/toolwindow/pullrequestlist/columns/NumberColumnInfo.java
package com.duxtinto.intellij.plugin.github.codereviews.ui.toolwindow.pullrequestlist.columns;

import com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests.PullRequestEntity;
import com.intellij.util.ui.ColumnInfo;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;
import javax.swing.table.DefaultTableCellRenderer;
import javax.swing.table.TableCellRenderer;
import java.awt.*;

public class NumberColumnInfo extends ColumnInfo<PullRequestEntity, Long> {
    public NumberColumnInfo() {
        super("number");
    }

    @Nullable
    @Override
    public Long valueOf(PullRequestEntity pullRequest) {
        return pullRequest.getNumber();
    }

    @Nullable
    @Override
    public TableCellRenderer getRenderer(PullRequestEntity pullRequest) {
        return new DefaultTableCellRenderer() {
            @Override
            public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int column) {
                JLabel cellContent = (JLabel) super.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, column);
                cellContent.setText("<html><a href=\"\">"+String.valueOf(value)+"</a></html>");
                return cellContent;
            }
        };
    }
}

File: src/main/kotlin/com/duxtinto/intellij/plugin/github/codereviews/presentation/pullrequestlist/columns/NumberColumnInfo.kt
package com.duxtinto.intellij.plugin.github.codereviews.presentation.pullrequestlist.columns

import com.duxtinto.intellij.plugin.github.codereviews.domain.pullrequests.PullRequestEntity
import com.intellij.util.ui.ColumnInfo

import javax.swing.*
import javax.swing.table.DefaultTableCellRenderer
import javax.swing.table.TableCellRenderer
import java.awt.*

class NumberColumnInfo : ColumnInfo<PullRequestEntity, Long>("number") {

    override fun valueOf(pullRequest: PullRequestEntity): Long? {
        return pullRequest.number
    }

    override fun getRenderer(pullRequest: PullRequestEntity?): TableCellRenderer? {
        return object : DefaultTableCellRenderer() {
            override fun getTableCellRendererComponent(table: JTable?, value: Any, isSelected: Boolean, hasFocus: Boolean, row: Int, column: Int): Component {
                val cellContent = super.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, column) as JLabel
                cellContent.text = "<html><a href=\"\">" + value.toString() + "</a></html>"
                return cellContent
            }
        }
    }
}
